chia/types/eligible_coin_spends.py

```diff
@@ -233,26 +233,34 @@ async def process_fast_forward_spends(
         get_unspent_lineage_info_for_puzzle_hash: Callable[[bytes32], Awaitable[Optional[UnspentLineageInfo]]],
         height: uint32,
         constants: ConsensusConstants,
-    ) -> None:
+    ) -> InternalMempoolItem:
         """
-        Provides the caller with an in-place internal mempool item that has a
-        proper state of fast forwarded coin spends and additions starting from
+        Provides the caller with an internal mempool item that has a proper
+        state of fast forwarded coin spends and additions starting from
         the most recent unspent versions of the related singleton spends.
 
         Args:
-            mempool_item: in-out parameter for the internal mempool item to process
+            mempool_item: The internal mempool item to process
             get_unspent_lineage_info_for_puzzle_hash: to lookup the most recent
                 version of the singleton from the coin store
             constants: needed in order to refresh the mempool item if needed
             height: needed in order to refresh the mempool item if needed
 
+        Returns:
+            The resulting mempool item after fast forwarding
+
         Raises:
             If a fast forward cannot proceed, to prevent potential double spends
         """
+
+        # Let's first create a copy of the mempool item to work on and return.
+        # This way we avoid the possibility of propagating a modified version
+        # of this input item through the network.
+        resulting_item = dataclasses.replace(mempool_item)
```

Comment:
```suggestion
        mempool_item = dataclasses.replace(mempool_item)
```

This means you can make the patch smaller because you don't need to rename the variable you use below. It also ensures you don't accidentally refer to the original mempool item, but only the copy.

Reply:
That's reasonable. Done.